projects/python-contract-examples/smart_contracts/arc4_types/contract.py

```diff
@@ -0,0 +1,305 @@
+# pyright: reportMissingModuleSource=false
+import typing as t
+
+from algopy import (
+    ARC4Contract,
+    GlobalState,
+    String,
+    UInt64,
+    arc4,
+    urange,
+)
+from algopy.arc4 import abimethod
+
+
+class Arc4Types(ARC4Contract):
+
+    # example: ARC4_UINT64
+    @abimethod()
+    def add_arc4_uint64(self, a: arc4.UInt64, b: arc4.UInt64) -> arc4.UInt64:
+        """
+        Math operations (like a + b) are not supported on arc4.UInt64 types
+        since they are internally represented as byte arrays in the AVM.
+        Use the .native property to perform arithmetic operations.
+        """
+
+        # Use the native integers to perform arithmetic
+        c = a.native + b.native
+
+        return arc4.UInt64(c)
+
+    # example: ARC4_UINT64
+
+    # example: ARC4_UINTN
+    @abimethod()
+    def add_arc4_uint_n(
+        self, a: arc4.UInt8, b: arc4.UInt16, c: arc4.UInt32, d: arc4.UInt64
+    ) -> arc4.UInt64:
+        """
+        The encoding of arc4 integers will be smaller if it uses fewer bits.
+        Ultimately, they are all represented with native UInt64.
+        """
+        assert a.bytes.length == 1
+        assert b.bytes.length == 2
+        assert c.bytes.length == 4
+        assert d.bytes.length == 8
```

Comment:
Add comments for clarity.

```suggestion
        assert a.bytes.length == 1  # UInt8 = 1 byte
        assert b.bytes.length == 2  # UInt16 = 2 bytes
        assert c.bytes.length == 4  # UInt32 = 4 bytes
        assert d.bytes.length == 8  # UInt64 = 8 bytes
```